sdk/communication/AzureCommunication/Tests/ObjCCommunicationUserCredentialAsyncTests.m

```diff
@@ -0,0 +1,159 @@
+@end
+
+@implementation ObjCCommunicationUserCredentialAsyncTests
+
+- (void)setUp {
+    [super setUp];
+    
+    self.sampleToken = @"eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHAiOjMyNTAzNjgwMDAwfQ.9i7FNNHHJT8cOzo-yrAUJyBSfJ-tPPk2emcHavOEpWc";
+    self.sampleExpiredToken = @"eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHAiOjEwMH0.1h_scYkNp-G98-O4cW6KvfJZwiz54uJMyeDACE4nypg";
+    self.sampleTokenExpiry = 32503680000;
+    self.fetchTokenCallCount = 0;
+}
+
+- (void)test_ObjCRefreshTokenOnDemand_AsyncRefresh {
+    XCTestExpectation *expectation = [self expectationWithDescription:
+                                      @"RefreshTokenOnDemand_AsyncRefresh"];
+    __weak ObjCCommunicationUserCredentialAsyncTests *weakSelf = self;
+    
+    CommunicationUserCredential *credential = [[CommunicationUserCredential alloc]
+                                               initWithInitialToken:self.sampleExpiredToken
+                                               refreshProactively:NO
+                                               error:nil
+                                               tokenRefresher:
+                                               ^(void (^ _Nonnull block)
+                                                 (NSString * _Nullable token,
+                                                  NSError * _Nullable error)) {
+            weakSelf.fetchTokenCallCount += 1;
+            block(weakSelf.sampleToken, nil);
+    }];
+    
+    [credential tokenWithCompletionHandler:^(CommunicationAccessToken * _Nullable accessToken,
+                                             NSError * _Nullable error) {
+        XCTAssertNotNil(accessToken);
+        XCTAssertNil(error);
+        XCTAssertEqual(accessToken.token, weakSelf.sampleToken);
+        XCTAssertEqual(accessToken.expiresOn.timeIntervalSince1970, weakSelf.sampleTokenExpiry);
+        
+        [expectation fulfill];
+    }];
+    
+    [self waitForExpectations:@[expectation] timeout:5.0];
+}
+
+- (void)test_ObjCRefreshTokenProactively_TokenExpiringInOneMin {
+    XCTestExpectation *expectation = [self expectationWithDescription:
+                                      @"RefreshTokenProactively_TokenExpiringInOneMin"];
+    __weak ObjCCommunicationUserCredentialAsyncTests *weakSelf = self;
+    
+    NSString *token = [self generateTokenValidForMinutes: 1];
+    CommunicationUserCredential *credential = [[CommunicationUserCredential alloc]
+                                               initWithInitialToken:token
+                                               refreshProactively:YES
+                                               error:nil
+                                               tokenRefresher:
+                                               ^(void (^ block)
+                                                 (NSString * _Nullable newToken,
+                                                  NSError * _Nullable error)) {
+        weakSelf.fetchTokenCallCount += 1;
```

Comment:
Do we have assert to validate the value of `weakSelf.fetchTokenCallCount` ?

Reply:
Yes this is being asserted in line 101 in the completion block